### PR TITLE
Rogue mode for commit only

### LINKED ISSUE #399: rogue mode for commit

### PR DESCRIPTION
**Pull request by ashish10677**
# Description

You can specify `--rogue --rogueMode commit` to the vote command to report wrong values during the commit state.

Fixes #399 

## Type of change

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

- [x] Tested on local hardhat

**Test Configuration**:
* Contracts version: v0.1.76
* Hardware: Macbook m1 air

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: cmd/cmd-utils_test.go [hunk 1]
```diff
@@ -134,7 +134,7 @@ func TestWaitForAppropriateState(t *testing.T) {
 		epochOrStateErr error
 		action          string
 		states          int
-		contains        bool
+		//contains        bool
```
Comment (SkandaBhat) [inline cmd/cmd-utils_test.go:137]: Why is this commented?

File: cmd/cmd-utils_test.go [hunk 2]
```diff
@@ -145,35 +145,35 @@ func TestWaitForAppropriateState(t *testing.T) {
...
-				epoch:    4,
-				state:    1,
-				action:   "reveal",
-				states:   1,
-				contains: true,
+				epoch:  4,
+				state:  1,
+				action: "reveal",
+				states: 1,
+				//contains: true,
```
Comment (SkandaBhat) [inline cmd/cmd-utils_test.go:152]: Remove comment

File: cmd/cmd-utils_test.go [hunk 3]
```diff
@@ -145,35 +145,35 @@ func TestWaitForAppropriateState(t *testing.T) {
...
-				epoch:    4,
-				state:    0,
-				action:   "commit",
-				states:   0,
-				contains: true,
+				epoch:  4,
+				state:  0,
+				action: "commit",
+				states: 0,
+				//contains: true,
```
Comment (SkandaBhat) [inline cmd/cmd-utils_test.go:164]: Remove comment

File: cmd/cmd-utils_test.go [hunk 4]
```diff
@@ -145,35 +145,35 @@ func TestWaitForAppropriateState(t *testing.T) {
...
-				epoch:    4,
-				state:    3,
-				action:   "dispute",
-				states:   3,
-				contains: true,
+				epoch:  4,
+				state:  3,
+				action: "dispute",
+				states: 3,
+				//contains: true,
```
Comment (SkandaBhat) [inline cmd/cmd-utils_test.go:176]: Remove comment

File: cmd/cmd-utils_test.go [hunk 5]
```diff
@@ -184,7 +184,7 @@ func TestWaitForAppropriateState(t *testing.T) {
 				epochOrStateErr: errors.New("error in fetching epoch and state"),
 				action:          "commit",
 				states:          0,
-				contains:        true,
+				//contains:        true,
```
Comment (SkandaBhat) [inline cmd/cmd-utils_test.go:187]: Remove comment

File: cmd/cmd-utils_test.go [hunk 6]
```diff
@@ -195,10 +195,10 @@ func TestWaitForAppropriateState(t *testing.T) {
 			GetEpochAndStateMock = func(*ethclient.Client, string, UtilsStruct) (uint32, int64, error) {
 				return tt.args.epoch, tt.args.state, tt.args.epochOrStateErr
 			}
-
-			ContainsMock = func([]int, int) bool {
-				return tt.args.contains
-			}
+			//
```
Comment (SkandaBhat) [inline cmd/cmd-utils_test.go:198]: Remove comment

File: cmd/vote.go [hunk 7]
```diff
@@ -44,7 +44,12 @@ Example:
 		utils.CheckError("Error in fetching config details: ", err)
 
 		password := utils.AssignPassword(cmd.Flags())
-		rogueMode, _ := cmd.Flags().GetBool("rogue")
+		isRogue, _ := cmd.Flags().GetBool("rogue")
+		rogueMode, _ := cmd.Flags().GetStringSlice("rogueMode")
+		rogue := types.Rogue{
```
Comment (SkandaBhat) [inline cmd/vote.go:49]: Call it rogueObject or rogueData. Just rogue makes it seem like a bool.

Comment (ashish10677) [reply cmd/vote.go:49]: Ok

**Review by SkandaBhat — CHANGES_REQUESTED** [review]
.

File: cmd/cmd-utils_test.go (hunk 1 above)
Comment (ashish10677) [reply cmd/cmd-utils_test.go:137]: I'll remove the comment. `contains` is no longer required.

**Review by SkandaBhat — APPROVED** [review]

**Pull request merged**